python/ray/tune/ray_trial_executor.py

```diff
@@ -108,19 +108,21 @@ def _setup_remote_runner(self, trial, reuse_allowed):
 
         def logger_creator(config):
             # Set the working dir in the remote process, for user file writes
-            if not os.path.exists(remote_logdir):
-                os.makedirs(remote_logdir)
+            trial_dir_schema.mkdir()
             if not ray.worker._mode() == ray.worker.LOCAL_MODE:
-                os.chdir(remote_logdir)
-            return NoopLogger(config, remote_logdir)
+                os.chdir(trial_dir_schema.remote_logdir)
+            return NoopLogger(config, trial_dir_schema.remote_logdir)
 
         # Clear the Trial's location (to be updated later on result)
         # since we don't know where the remote runner is placed.
         trial.set_location(Location())
         logger.info("Trial %s: Setting up new remote runner.", trial)
         # Logging for trials is handled centrally by TrialRunner, so
         # configure the remote runner to use a noop-logger.
-        return cls.remote(config=trial.config, logger_creator=logger_creator)
+        runner = cls.remote(config=trial.config, logger_creator=logger_creator)
+        # Initialize the checkpoint directory.
+        runner._init_checkpoint_dir.remote(trial.checkpoint_dir)
```

Comment:
Why not just move this into logger_creator? 

After all, the trial_dir_schema is already in the logger creator.